android/cli/commands/_cleanModule.js

```diff
@@ -0,0 +1,37 @@
+/**
+* Android module clean command.
+*
+* @module cli/_cleanModule
+*
+* @copyright
+* Copyright (c) 2018 by Appcelerator, Inc. All Rights Reserved.
+*
+* @license
+* Licensed under the terms of the Apache Public License
+* Please see the LICENSE included with this distribution for details.
+*/
+
+'use strict';
+const path = require('path');
+const fs = require('fs-extra');
+const appc = require('node-appc');
+const __ = appc.i18n(__dirname).__;
+
+// TODO Do we need a validate function?
+
+exports.run = function run(logger, config, cli, finished) {
+	const projectDir = cli.argv['project-dir'];
+
+	const toDelete = [ 'build', 'dist', 'libs', 'java-sources.txt' ];
```

Comment:
I think the build command already handles the recreation, but "build" has to exist and "libs" has to include the used architectures, e.g `arm64-v8a`. Also, [many](https://github.com/benbahrenburg/benCoding.AlarmManager/tree/master/dist) devs curate a list of archives in the `dist` directory that will be flushed out now

Reply:
Android goes a little nutty if `libs` already exists, sometimes. I think it tries to do an architecture sanity check, but I'm not sure why because that's an output directory for the static libraries it generates during he build.

I tested cleaning and building a couple native modules, and it handles recreating the `build` dir.

As far as `dist` goes, I get that some devs keep the generated zips there and check them into SCM, but that's not exactly a great practice. The zips should be published to S3 or Github releases or something, not checked into source.

Reply:
I'd still rather not want to break this in a minor version and don't see the advantage of deleting them before, as they do not cause any issues right now, different to the other cleaned directories. Maybe move that change to 8.0.0 instead.

Reply:
I struggle to see how this is a breaking change. If a developer has a not-so-great practice of checking in generated zips to their SCM and the new module clean command will delete that directory, the file will still be in their SCM. This command will not "checkin" the deletion of the zip. And even if the user did check the deletion into SCM, the file would still be in their SCM history to restore.

The `dist` folder is generated output folder that should be cleaned before a clean build.

Reply:
I only see the following JIRA ticket incoming:
> "I create super duper nice module packages since 6 years and now I lost all my clients because the module zips are gone. Why do you do that?" 
If we can defend that, let's do it.